core/src/main/scala/org/apache/spark/sql/eventhubs/EventHubsSourceProvider.scala

```diff
@@ -155,7 +155,10 @@ private[sql] object EventHubsSourceProvider extends Serializable {
         StructField("sequenceNumber", LongType),
         StructField("enqueuedTime", TimestampType),
         StructField("publisher", StringType),
-        StructField("partitionKey", StringType)
+        StructField("partitionKey", StringType),
+        StructField("properties",
+                    MapType(StringType, StringType, valueContainsNull = true),
```

Comment:
is there a reason that you had to change from what EventData has: `Map<String, Object> getProperties();` !!?

One advantage with going with `Map<String,Object>` is users can send any type for `value` and transform it per their need.

Reply:
`MapType` requires a `DataType`. `AnyDataType` is a `AbstractDataType` - seems `StringType` was the best fit.